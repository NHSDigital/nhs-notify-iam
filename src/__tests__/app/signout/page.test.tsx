import { ReactNode, ComponentType } from 'react';
import { render, screen } from '@testing-library/react';
import SignoutPage from '../../../app/signout/page';

jest.mock('@aws-amplify/auth', () => ({
  signOut: () => Promise.resolve(),
}));

jest.mock('../../../components/molecules/Redirect/Redirect', () => ({
  Redirect: () => <p>redirect</p>,
}));

jest.mock('@aws-amplify/ui-react', () => ({
  Authenticator: {
    Provider: ({ children }: { children: ReactNode }) => children,
  },
  withAuthenticator: (Component: ComponentType) => Component,
}));

test('MockSignoutPage', async () => {
  const container = render(<SignoutPage />);

  await screen.findByText('redirect');

  expect(container.asFragment()).toMatchSnapshot();
});
