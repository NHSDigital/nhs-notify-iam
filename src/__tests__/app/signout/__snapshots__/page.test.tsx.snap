// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signout Page calls signOut when user is authenticated 1`] = `
<DocumentFragment>
  <p>
    Signing out
  </p>
</DocumentFragment>
`;

exports[`Signout Page does not call signOut when user is unauthenticated 1`] = `
<DocumentFragment>
  <p>
    Signed out
  </p>
</DocumentFragment>
`;
